Sources/Plugins/AttachmentManager/AttachmentManager.swift

```diff
@@ -121,6 +121,17 @@ open class AttachmentManager: NSObject, InputPlugin {
         })
     }
     
+    /// Performs an animated removal of all attachments
+    open func removeAllAttachments() {
+        let attachments = self.attachments
+        self.attachments.removeAll()
+        self.reloadData()
+        for i in 0..<attachments.count {
+            self.delegate?.attachmentManager(self, didRemove: attachments[i], at: i)
```

Comment:
The only method I saw to notify the delegate is this one.

Reply:
Sorry to come back here after year and half, but I would propose to add new optional delegate method `attachmentManagerDidRemoveAllAttachments(self)` - we will be able to simply remove all the mess above, just remove all attachments, reload data, call the delegate without for cycle and that's it. It will be completely new API that won't interact with existing delegate API.